### ISSUE TITLE
`sigma0` should be `0.13`

### ISSUE THREAD

**Issue by OverLordGoldDragon** [kymatio/kymatio#789]
AFAIK 0.1 is a heuristic which aims to allow subsampling by `T` by setting `sigma=0.1/T`. It is, however, overly conservative, according to `criterion_amplitude=1e-3`:

```python
from kymatio.scattering1d.filter_bank import gauss_1d, compute_temporal_support
T = 128
phi = gauss_1d(16384, 0.1 / T)
f_at_subsampling = len(phi)//2//T
f_halfsupport = compute_temporal_support(ifft(phi), criterion_amplitude=1e-3)
print(f_halfsupport, f_at_subsampling)
```
```
49 64
```
vs `0.13 / T`:
```
63 64
```
`criterion_amplitude` conditions the extent of decay upon which boundary effects, i.e. tail contributions, are negligible. Hence, if contributions past `1e-3` are deemed negligible, then we have ~lossless subsampling - and `0.1` obtaining `49` means we subsample much less (by 30%) than we safely could.

It's a simple yet significant change by which the wavelets would also be affected (since `J` is also defined per `sigma0`) and will require changing some tests and maybe docs, so I leave the idea here for now.

**Comment by lostanlen** [comment]
Hi! If anything, doesn't your experiment suggest that we should have one parameter, not two, to govern this?
Why not specify `criterion_amplitude` and let Kymatio figure out that the corresponding sigma for Gauss should be (around)  `0.13` ? Do we need a hard-coded constant at all?

**Comment by OverLordGoldDragon** [comment]
Ideally, yes, exact `sigma` can be determined empirically by iterative methods, and it will depend on: `N, T, criterion_amplitude`. However, past a certain `N`, all `T` are satisfied with the same `sigma`. Differences become noticeable for `N < 128`, which is the case with JTFS (for `F`), but it's still mostly acceptable.

Since `0.13` remains a close fit for all `N`, a trivial approach is to sample several values around it and pick the best fit. But a no-brainer that can be merged sooner than later, for starters, is `sigma0=0.13` - the only challenging part is adjusting whatever tests this'll affect.

**Comment by lostanlen** [comment]
I don't get it. What happens if i set `criterion_amplitude=1e-6` ?

**Comment by OverLordGoldDragon** [comment]
Then the distance from peak to first point to decay below `1e-6 * max(phi)` will be the "support". This means much less subsampling (excessively).

**Comment by lostanlen** [comment]
ah, so, `criterion_amplitude` is meant to affect subsampling but not the design of the wavelet itself, in a continuous-time continuous-frequency setting.
then i don't see the use in modifying the shape of the wavelet so that this `halfsupport` matches a power of two (under the `Q=1` default). There is nothing special with this `1e-3` value, it's just an ad hoc constant. (and besides, it is modifiable!)

If anything, i could be convinced to modify `sigma0` by a reasoning on `Q` and the Littlewood-Paley sum in a continuous-time continuous-frequency setting; not by an appeal to the output of `compute_temporal_support`.

**Comment by lostanlen** [comment]
update: `sigma0` relates to the low-pass filter, not the band-pass filters. (am i right?)
so Littlewood-Paley doesn't help us decide here. my bad

so perhaps we could redesign this filter, after all. i imagine than increasing from `0.1` to `0.13` would smooth `U` a little less when mapping to `S`,  as some (acceptable) cost in aliasing?

**Comment by OverLordGoldDragon** [comment]
`criterion_amplitude` affects:
 
 - how `T` translates to `sigma` for `phi`
 - how `sigma` maps to `j` for `psi`
 - how `J` determines support of largest wavelet

There is no effect on shape.

<img src="https://user-images.githubusercontent.com/16495490/147890529-898f8493-01fa-4073-85a9-2c5505f5c2e7.png" width="400">

In terms of filters, changing `sigma0` would affect:

 1. width of `phi`
 2. width of largest `psi`, but not change width of any particular `psi`
 3. per 2, it will also affect the total number of wavelets: higher `sigma0` will reduce it since it reduces the largest permitted support (`sigma_min = sigma0 / 2**J`) and hence the number of filters we can fit via CQT

**Comment by lostanlen** [comment]
> In terms of filters, changing sigma0 would affect:
> 
>     width of phi
>     width of largest psi, but not change width of any particular psi
>     per 2, it will also affect the total number of wavelets:

LOL, where, clearly this is the root of the problem. The API is too opaque. We should have never let `criterion_amplitude` and `sigma0` be overriden by the user.

**Comment by OverLordGoldDragon** [comment]
> We should have never let criterion_amplitude and sigma0 be overriden by the user.

Well, it's not

**Comment by OverLordGoldDragon** [comment]
So correction,

> how sigma maps to j for psi

isn't true, but it should be. I took it so for granted I completely forgot about `alpha`.

**Comment by lostanlen** [comment]
Can you summarize? What is the role of `sigma0` in all this, and why should it be tunable, as opposed to derived from T?
Can't we fully specify a filterbank (both psis and phi) from the knowledge of `T`, `Q`, `J` ?

IIUC this value of `sigma0=0.1` what chosen heuristically so as to design `phi` from `T`. But now you want to attach a precise meaning to `T` in terms of invariance property (i support this idea!): hence `sigma0` should no longer be tunable, right? Meaning the constant 0.13 should somehow be hard-coded in Kymatio, and not be a loose parameter.

I must admit i still don't understand how this interacts with padding, subsampling and criterion_amplitude. Are you saying that the amount of empirical invariance you measure depends itself on boundary effects, which itself depends on how much we pad? So you want to redesign the filters to account for those boundary effects and better satisfy the required invariance T?
